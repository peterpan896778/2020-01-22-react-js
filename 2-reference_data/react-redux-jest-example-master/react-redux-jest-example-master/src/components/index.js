export { default as Button } from './Button';
export { default as Counter } from './Counter';
export { default as Logo } from './Logo';
