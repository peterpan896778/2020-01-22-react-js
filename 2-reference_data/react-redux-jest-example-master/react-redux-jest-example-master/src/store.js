import { createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import reducer from 'src/reducers';

const store = createStore(
  reducer,
  composeWithDevTools(),
);

export default store;
