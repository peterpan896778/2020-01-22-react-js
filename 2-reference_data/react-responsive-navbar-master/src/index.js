import ResponsiveNavbar from './responsive-navbar.component';

export default ResponsiveNavbar;
