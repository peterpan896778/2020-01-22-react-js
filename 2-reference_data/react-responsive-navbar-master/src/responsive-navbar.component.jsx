/* eslint-disable react/no-find-dom-node */
import React from 'react';
import PropTypes from 'prop-types';
import { FloatingSelect } from '@opuscapita/react-floating-select';
import { OverlayTrigger, Tooltip } from 'react-bootstrap';
import { debounce } from 'debounce';

import './responsive-navbar.scss';

export default class ResponsiveNavbar extends React.PureComponent {
  static propTypes = {
    id: PropTypes.string,
    className: PropTypes.string,
    showNavItemBorder: PropTypes.bool,
    showNavItemTooltip: PropTypes.bool,
    tooltipDelay: PropTypes.number,
    fontSize: PropTypes.string,
    fontWeight: PropTypes.string,
    placeholder: PropTypes.string,
    activeKey: PropTypes.oneOfType([PropTypes.shape({}), PropTypes.number]).isRequired,
    list: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.node,
      ]).isRequired,
      href: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    })).isRequired,
    onSelect: PropTypes.func,
    height: PropTypes.string,
    componentLeft: PropTypes.node,
    componentRight: PropTypes.node,
  }

  static defaultProps = {
    id: 'responsive-navbar',
    className: '',
    onSelect: () => {},
    showNavItemBorder: false,
    showNavItemTooltip: true,
    tooltipDelay: 2000,
    fontSize: 'inherit',
    fontWeight: 'inherit',
    placeholder: 'more...',
    height: '40px',
    componentLeft: null,
    componentRight: null,
  }

  constructor(props) {
    super(props);
    this.itemWidths = []; // store item widths here, they don't change
  }

  state = {
    isSelectVisible: false,
    lastVisibleItemIndex: -2,
  };

  componentDidMount() {
    window.addEventListener('resize', this.handleResize);
    window.addEventListener('orientationchange', this.refreshLastVisibleItem); // for mobile support
    this.refreshLastVisibleItem();
  }

  componentDidUpdate(prevProps, prevState) {
    // Refresh visible items if values change
    if (
      this.state.isSelectVisible !== prevState.isSelectVisible ||
      this.state.lastVisibleItemIndex !== prevState.lastVisibleItemIndex
    ) {
      this.refreshLastVisibleItem();
    }
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.handleResize);
    window.removeEventListener('orientationchange', this.refreshLastVisibleItem); // for mobile support
  }

  getLastVisibleItemIndex = () => {
    const navBarWidth = this.navbarContainerRef ? this.navbarContainerRef.offsetWidth : 0;
    const selectWidth = this.selectContainerRef ? this.selectContainerRef.offsetWidth : 0;
    const componentLeftWidth = this.componentLeftContainerRef ? this.componentLeftContainerRef.offsetWidth : 0; // eslint-disable-line
    const componentRightWidth = this.componentRightContainerRef ? this.componentRightContainerRef.offsetWidth : 0; // eslint-disable-line

    let remainingWidth = navBarWidth - selectWidth - componentLeftWidth - componentRightWidth;
    let lastVisible = 0;

    for (let i = 0; i < this.props.list.length; i += 1) {
      remainingWidth -= this.itemWidths[i];
      if (remainingWidth < 0) {
        lastVisible -= 1;
        break;
      }
      lastVisible += 1;
    }

    return lastVisible;
  }

  handleResize = () => debounce(this.refreshLastVisibleItem(), 300);

  refreshLastVisibleItem = () => {
    const lastVisibleItemIndex = this.getLastVisibleItemIndex();
    if (this.state.lastVisibleItemIndex !== lastVisibleItemIndex) {
      this.setState({
        isSelectVisible: this.props.list.length > (lastVisibleItemIndex + 1),
        lastVisibleItemIndex,
      });
    }
  }

  tooltipWrapper = (node, index, tooltipContent) => {
    const tooltip = <Tooltip id="tooltip">{tooltipContent}</Tooltip>;
    return !this.props.showNavItemTooltip ? node :
    <OverlayTrigger placement="bottom" key={index} overlay={tooltip} delayShow={this.props.tooltipDelay}>
      {node}
    </OverlayTrigger>;
  }

  // Handle react-select onChange
  handleOnChange = (value) => {
    this.props.onSelect(value);
  }

  // Handle navbar onClick
  handleOnClick = href => () => {
    this.props.onSelect(href);
  }

  // Render navbar item
  navbarItem = (item, index, className) => {
    const {
      activeKey,
      fontWeight,
      fontSize,
      height,
    } = this.props;

    // resolve activeKeyIndex
    let activeKeyIndex = activeKey;
    if (typeof activeKey === 'object') {
      activeKeyIndex = this.activeItemIndex(activeKey);
    }
    return (
      <button
        className={index === activeKeyIndex ? `${className} selected-border` : `${className}`}
        style={{ fontWeight, fontSize, minHeight: height }}
        id={item.id || `navItem${String(index)}`}
        key={item.id || `navitem${String(index)}`}
        onClick={this.handleOnClick(item.href)}
        ref={(r) => {
          if (r && !this.itemWidths[index]) this.itemWidths[index] = r.offsetWidth;
        }}
      >
        <span className="responsive-navbar-item-text">{item.name}</span>
      </button>
    );
  }

  doLineCount = () => {
    const { list } = this.props;
    return list.some(item => typeof (item.name) !== 'string');
  }

  // Render navbar
  navbar = () => {
    const {
      id,
      className,
      list,
      showNavItemBorder,
      height,
    } = this.props;
    const visibleList = this.state.lastVisibleItemIndex > -2 ?
      list.slice(0, this.state.lastVisibleItemIndex + 1) :
      list;
    const itemClassName = showNavItemBorder ?
      'responsive-navbar-item inactive-border' :
      'responsive-navbar-item no-item-border';
    const items = visibleList.map((item, index) => (
      this.tooltipWrapper(this.navbarItem(item, index, itemClassName), index, item.name)
    ));
    const lineCount = this.doLineCount();
    const navbarStyle = {
      minHeight: height,
    };
    if (height.slice(-2) === 'px' && lineCount) {
      const heightPx = parseInt(height.slice(0, -2), 10);
      navbarStyle.lineHeight = `${(heightPx - 4)}px`;
    }
    return (
      <div
        id={`${id}-container`}
        ref={(r) => { this.navbarContainerRef = r; }}
        className={`responsive-navbar-container ${className}`}
        style={navbarStyle}
      >
        {items}
        {this.combobox()}
        {this.componentLeft()}
        {this.componentRight()}
      </div>
    );
  }

  resolveActiveItemFromOptions = (selectOptions) => {
    const { activeKey } = this.props;
    let activeItem = selectOptions.find(opts => opts.value === activeKey);
    if (!activeItem) {
      activeItem = selectOptions.find(opts => opts.value === activeKey.value);
    }
    return activeItem;
  }

  activeItemIndex = (activeItem) => {
    const { list } = this.props;
    if (!activeItem) return null;
    return list.findIndex(item => item.href === activeItem.value);
  }

  // Render combobox, when all items do not fit
  combobox = () => {
    const {
      id,
      list,
      fontSize,
      fontWeight,
      placeholder,
      showNavItemBorder,
    } = this.props;
    if (!this.state.isSelectVisible) {
      // return null if all nav items are visible
      return null;
    }

    // slice nav items list and show invisible items in the combobox
    const navList = this.state.lastVisibleItemIndex > -2 ?
      list.slice(this.state.lastVisibleItemIndex + 1) : list;
    const selectOptions = navList.map(item => ({
      value: item.href,
      label: item.name,
    }));
    const lineCountNeeded = this.doLineCount();
    const customLineCount = lineCountNeeded ? 'line-count' : '';
    const customBorderClass = lineCountNeeded ? 'selected-border line-count' : 'selected-border';
    const customInactiveBorder = lineCountNeeded ? 'inactive-border line-count' : 'inactive-border';
    const inactiveBorder = showNavItemBorder ? customInactiveBorder : customLineCount;
    // Resolve activeItem
    const activeItem = this.resolveActiveItemFromOptions(selectOptions);
    const activeItemIndex = this.activeItemIndex(activeItem);
    const borderClass = activeItemIndex >= (this.state.lastVisibleItemIndex + 1) ? customBorderClass : inactiveBorder; // eslint-disable-line

    return (
      <div
        id={`${id}-select`}
        className={`responsive-navbar-select ${borderClass}`}
        style={{ fontWeight, fontSize }}
        ref={(r) => { this.selectContainerRef = r; }}
      >
        <FloatingSelect
          name={`${id}-select-component`}
          value={activeItem || ''}
          isClearable={false}
          placeholder={placeholder}
          options={selectOptions}
          onChange={this.handleOnChange}
        />
      </div>
    );
  }

  // Render custom left side component
  componentLeft = () => {
    const { componentLeft } = this.props;
    if (!componentLeft) return null;
    return (
      <div
        className="responsive-navbar-container-left"
        ref={(r) => { this.componentLeftContainerRef = r; }}
      >
        { componentLeft }
      </div>
    );
  }

  // Render custom right side component
  componentRight = () => {
    const { componentRight } = this.props;
    if (!componentRight) return null;
    return (
      <div
        className="responsive-navbar-container-right"
        ref={(r) => { this.componentRightContainerRef = r; }}
      >
        { componentRight }
      </div>
    );
  }

  render() {
    return this.navbar();
  }
}
