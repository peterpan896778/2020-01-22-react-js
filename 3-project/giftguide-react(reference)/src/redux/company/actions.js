import {
  GET_COMPANIES,
  ADD_COMPANY,
  ADD_COMPANY_SUCCESS,
  GET_COMPANY,
  GET_COMPANY_SUCCESS,
  CREATE_COLUMN,
  CREATE_COLUMN_SUCCESS,
  SET_ACTIVE,
  UPDATE_COMPANY
} from "../../constants/actionTypes";

type CompanyAction = {
  type: string,
  payload: {} | string
};

export const addCompany = (
  company: string,
): CompanyAction => ({
  type: ADD_COMPANY,
  payload: { company }
});

export const addCompanySuccess = (companies: []): CompanyAction => ({
  type: ADD_COMPANY_SUCCESS,
  payload: companies
});

export const editCompanySuccess = (company: []): CompanyAction => ({
  type: GET_COMPANY_SUCCESS,
  payload: company
});

export const getCompanies = (): CompanyAction => ({
  type: GET_COMPANIES
});

export const getCompany = (id: string): CompanyAction => ({
  type: GET_COMPANY,
  payload: id
});

export const createColumn = (id: string): CompanyAction => ({
  type: CREATE_COLUMN
});

export const createColumnSuccess = (): CompanyAction => ({
  type: CREATE_COLUMN_SUCCESS
});

export const setActive = (active: string): CompanyAction => ({
  type: SET_ACTIVE,
  payload: active
});

export const updateCompany = (company, id): CompanyAction =>({
  type: UPDATE_COMPANY,
  payload: { id, company }
});
