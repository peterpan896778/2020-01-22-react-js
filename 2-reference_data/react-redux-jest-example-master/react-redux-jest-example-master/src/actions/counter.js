import { INCREMENT, DECREMENT } from 'src/constants';

export const increment = () => ({
  type: INCREMENT,
});

export const decrement = () => ({
  type: DECREMENT,
});
