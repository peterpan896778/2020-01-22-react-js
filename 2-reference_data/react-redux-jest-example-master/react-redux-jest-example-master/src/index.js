import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { App } from 'src/containers';

import store from './store';
import registerServiceWorker from './config/registerServiceWorker';

ReactDOM.render(
  <Provider store={store}>
    <App />
  </Provider>,
  document.getElementById('root'),
);

registerServiceWorker();
