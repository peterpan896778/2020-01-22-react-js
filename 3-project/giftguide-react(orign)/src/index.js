import React from "react";
import ReactDOM from "react-dom";
import { Provider } from "react-redux";
import App from "./components/App/App";
import store from "./stores";
import { createBrowserHistory as createHistory } from "history";

import "../node_modules/bootstrap/dist/css/bootstrap.min.css";
import * as serviceWorker from "./serviceWorker";

const history = createHistory();

ReactDOM.render(
  <Provider store={store}>
    <App history={history} />
  </Provider>,
  document.getElementById("root")
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: https://bit.ly/CRA-PWA
serviceWorker.unregister();

// import App from "./App";
// import { configureStore } from "./redux/store";

// export const store = configureStore();
