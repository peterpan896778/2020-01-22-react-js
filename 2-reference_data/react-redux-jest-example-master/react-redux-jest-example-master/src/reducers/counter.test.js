import reducer from 'src/reducers/counter';
import { increment, decrement } from 'src/actions';

describe('counter reducer', () => {
  test('counter should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  test('counter/INCREMENT \'10\' to  \'11\'', () => {
    expect(reducer({ current: 10 }, increment())).toMatchSnapshot();
  });

  test('counter/DECREMENT \'5\' to  \'4\'', () => {
    expect(reducer({ current: 5 }, decrement())).toMatchSnapshot();
  });
});
