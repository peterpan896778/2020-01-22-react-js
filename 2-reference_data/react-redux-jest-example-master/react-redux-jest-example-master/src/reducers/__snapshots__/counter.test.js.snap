// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter reducer counter should return the initial state 1`] = `
Object {
  "current": 0,
}
`;

exports[`counter reducer counter/DECREMENT '5' to  '4' 1`] = `
Object {
  "current": 4,
}
`;

exports[`counter reducer counter/INCREMENT '10' to  '11' 1`] = `
Object {
  "current": 11,
}
`;
